common.go

```diff
@@ -78,8 +93,142 @@ type PaymentMethodDetails struct {
 	AdditionalDetails AdditionalDetails `json:"additional_details,omitempty"`
 }
 
-// PaymentMethodHref wraps PaymentMethod with associated href.
-type PaymentMethodHref struct {
-	Href          string         `json:"href"`
-	PaymentMethod *PaymentMethod `json:"payment_method"`
+// This object represents the configuration of the provider that handled the transaction,
+// as defined in your PaymentsOS Control Center account.
+// Note that the object does not include your provider authentication credentials.
+type ProviderConfiguration struct {
+	ID          string      `json:"id"`
+	Name        string      `json:"name"`
+	Description string      `json:"description"`
+	Created     json.Number `json:"created"`
+	Modified    json.Number `json:"modified"`
+	ProviderID  string      `json:"provider_id"`
+	Type        string      `json:"type"`
+	AccountID   string      `json:"account_id"`
+	Href        string      `json:"href"`
+}
+
+// The line items of the order.
+type OrderLineItem struct {
+	Name      string `json:"name"`
+	ID        string `json:"id"`
+	Quantity  int    `json:"quantity"`
+	UnitPrice int64  `json:"unit_price"`
+}
+
+// Details of the order. Note that order fields required for level 2 and 3 data, must be passed separately
+// in a Create Capture request within a level_2_3 object (fields passed in the order object only are not recognized as level 2 and 3 data fields).
+type Order struct {
+	ID                string            `json:"id"`
+	AdditionalDetails AdditionalDetails `json:"additional_details,omitempty"`
+	LineItems         []OrderLineItem   `json:"line_items"`
+	TaxAmount         int64             `json:"tax_amount"`
+	TaxPercentage     decimal.Decimal   `json:"tax_percentage"`
+}
+
+type DecodedJSON map[string]interface{}
```

Comment:
Do we need to support a case when provider raw response is not a json at all?

Reply:
According to the ZOOZ documentation for two fields:
```markdown

raw_response | string The raw response received. This field is for support and debugging purposes only. Beware: Do not use this field for any business logic in a production environment, since its structure may change.

```
and 
```markdown



object (provider_specific_data)  This object contains provider specific objects, which contain data in key-value format. Use the Bodybuilder to generate a sample request that includes a provider_specific_data object.
property name*object | property name* | object
property name* | object


```
In raw_response it can be not the JSON object but practically it's the JSON object and scripts of Leb has based that it's the JSON)

Reply:
If you want I can return the data as is if we can't unmarshal it as JSON, WDYTH?